File: internal/models/company.go
package models

import (
	"time"
)

// Company represents an organization within a tenant
type Company struct {
	ID                  string    `json:"id" db:"id"`
	TenantID            string    `json:"tenant_id" db:"tenant_id"`
	Name                string    `json:"name" db:"name"`
	Description         string    `json:"description" db:"description"`
	Status              string    `json:"status" db:"status"` // "active", "inactive", "suspended"
	IndustryType        string    `json:"industry_type" db:"industry_type"`
	EmployeeCount       *int      `json:"employee_count" db:"employee_count"`
	Website             *string   `json:"website" db:"website"`
	MaxProjects         int       `json:"max_projects" db:"max_projects"`
	MaxUsers            int       `json:"max_users" db:"max_users"`
	CurrentUserCount    int       `json:"current_user_count" db:"current_user_count"`
	CurrentProjectCount int       `json:"current_project_count" db:"current_project_count"`
	BillingEmail        string    `json:"billing_email" db:"billing_email"`
	BillingAddress      string    `json:"billing_address" db:"billing_address"`
	CreatedAt           time.Time `json:"created_at" db:"created_at"`
	UpdatedAt           time.Time `json:"updated_at" db:"updated_at"`
}

// Project represents a project under a company
type Project struct {
	ID               string     `json:"id" db:"id"`
	CompanyID        string     `json:"company_id" db:"company_id"`
	TenantID         string     `json:"tenant_id" db:"tenant_id"`
	Name             string     `json:"name" db:"name"`
	Description      string     `json:"description" db:"description"`
	Status           string     `json:"status" db:"status"`             // "active", "inactive", "archived"
	ProjectType      string     `json:"project_type" db:"project_type"` // "sales", "support", "marketing", "custom"
	MaxUsers         int        `json:"max_users" db:"max_users"`
	CurrentUserCount int        `json:"current_user_count" db:"current_user_count"`
	BudgetAllocated  float64    `json:"budget_allocated" db:"budget_allocated"`
	BudgetSpent      float64    `json:"budget_spent" db:"budget_spent"`
	StartDate        time.Time  `json:"start_date" db:"start_date"`
	EndDate          *time.Time `json:"end_date" db:"end_date"`
	CreatedAt        time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt        time.Time  `json:"updated_at" db:"updated_at"`
}

// CompanyMember represents a user's membership in a company
type CompanyMember struct {
	ID         string    `json:"id" db:"id"`
	CompanyID  string    `json:"company_id" db:"company_id"`
	UserID     int       `json:"user_id" db:"user_id"`
	TenantID   string    `json:"tenant_id" db:"tenant_id"`
	Role       string    `json:"role" db:"role"` // "owner", "admin", "manager", "member", "viewer"
	Department string    `json:"department" db:"department"`
	IsActive   bool      `json:"is_active" db:"is_active"`
	CreatedAt  time.Time `json:"created_at" db:"created_at"`
	UpdatedAt  time.Time `json:"updated_at" db:"updated_at"`
}

// ProjectMember represents a user's membership in a project
type ProjectMember struct {
	ID        string    `json:"id" db:"id"`
	ProjectID string    `json:"project_id" db:"project_id"`
	UserID    int       `json:"user_id" db:"user_id"`
	CompanyID string    `json:"company_id" db:"company_id"`
	TenantID  string    `json:"tenant_id" db:"tenant_id"`
	Role      string    `json:"role" db:"role"` // "lead", "member", "viewer", "analyst"
	JoinedAt  time.Time `json:"joined_at" db:"joined_at"`
	IsActive  bool      `json:"is_active" db:"is_active"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
	UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
}

// UserRole represents a role and its permissions (cross-company/project)
type UserRole struct {
	ID          string    `json:"id" db:"id"`
	TenantID    string    `json:"tenant_id" db:"tenant_id"`
	Name        string    `json:"name" db:"name"` // "admin", "accountant", "manager"
	Description string    `json:"description" db:"description"`
	Permissions string    `json:"permissions" db:"permissions"` // JSON array of permission strings
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

File: frontend/types/company.ts
export interface Company {
  id: string
  tenant_id: string
  name: string
  description: string
  status: 'active' | 'inactive' | 'suspended'
  industry_type: string
  employee_count?: number
  website?: string
  max_projects: number
  max_users: number
  current_user_count: number
  current_project_count: number
  billing_email: string
  billing_address: string
  created_at: string
  updated_at: string
}

export interface Project {
  id: string
  company_id: string
  tenant_id: string
  name: string
  description: string
  status: 'active' | 'inactive' | 'archived'
  project_type: 'sales' | 'support' | 'marketing' | 'custom'
  max_users: number
  current_user_count: number
  budget_allocated: number
  budget_spent: number
  start_date: string
  end_date?: string
  created_at: string
  updated_at: string
}

export interface CompanyMember {
  id: string
  company_id: string
  user_id: number
  tenant_id: string
  role: 'owner' | 'admin' | 'manager' | 'member' | 'viewer'
  department: string
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface ProjectMember {
  id: string
  project_id: string
  user_id: number
  company_id: string
  tenant_id: string
  role: 'lead' | 'member' | 'viewer' | 'analyst'
  joined_at: string
  is_active: boolean
  created_at: string
  updated_at: string
}
